plugin-api/src/main/java/org/sonar/api/security/Authenticator.java

```diff
@@ -43,15 +44,31 @@ public abstract class Authenticator {
   public static final class Context {
     private String username;
     private String password;
+    /**
+     * @deprecated since 9.16 use {@link #httpRequest} instead
+     */
+    @Deprecated(since = "9.16", forRemoval = true)
     private HttpServletRequest request;
+    private HttpRequest httpRequest;
 
+    /**
+     * @deprecated since 9.16 use {@link #Context(String, String, HttpRequest)} instead
+     */
+    @Deprecated(since = "9.16", forRemoval = true)
     public Context(@Nullable String username, @Nullable String password, HttpServletRequest request) {
       requireNonNull(request);
       this.request = request;
       this.username = username;
       this.password = password;
```

Comment:
Should I also add like
`this.httpRequest = new JavaxHttpRequest(request);`
On the deprecated constructor?